### PR TITLE
fix: [IAC-1288]: Fix evaluation status

### PR DESCRIPTION
**Pull request by willd0yle**
### Summary
Fix evaluation status when governance status is an empty string. A scenario occurs where there is governance data however the status has not been set yet.

This fix sets the status to Error if either iacm or governance has an error, warning if either has a warning otherwise it will be set to pass until the data has been updated.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    if (govData.status === EvaluationStatus.ERROR || iacmStatus === EvaluationStatus.ERROR) {
+      govData.status = EvaluationStatus.ERROR
+    } else if (govData.status === EvaluationStatus.WARNING || iacmStatus === EvaluationStatus.WARNING) {
+      govData.status = EvaluationStatus.WARNING
+    } else {
+      govData.status = EvaluationStatus.PASS
+    }
+  }
+
+  return govData
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts:102]: Minor improvement, but you could bail early if `IACM_ENABLED` is `false`.

```suggestion
  if (!IACM_ENABLED) return govData

  const { iacmData, iacmStatus } = formatIacmGovernanceData(data)
  if (iacmData && !isLoading) {
    govData.details = [...defaultTo(govData.details, []), ...iacmData]
  }

  if (iacmStatus !== govData.status) {
    if (govData.status === EvaluationStatus.ERROR || iacmStatus === EvaluationStatus.ERROR) {
      govData.status = EvaluationStatus.ERROR
    } else if (govData.status === EvaluationStatus.WARNING || iacmStatus === EvaluationStatus.WARNING) {
      govData.status = EvaluationStatus.WARNING
    } else {
      govData.status = EvaluationStatus.PASS
    }
  }

  return govData
```

Or just wrap this stuff
```suggestion
  if (IACM_ENABLED) {
    const { iacmData, iacmStatus } = formatIacmGovernanceData(data)
  
    if (iacmData && !isLoading) {
      govData.details = [...defaultTo(govData.details, []), ...iacmData]
    }

    if (iacmStatus !== govData.status) {
      if (govData.status === EvaluationStatus.ERROR || iacmStatus === EvaluationStatus.ERROR) {
        govData.status = EvaluationStatus.ERROR
      } else if (govData.status === EvaluationStatus.WARNING || iacmStatus === EvaluationStatus.WARNING) {
        govData.status = EvaluationStatus.WARNING
      } else {
        govData.status = EvaluationStatus.PASS
      }
    }
  }

  return govData
```

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        policySetId: defaultEmpty(policySet.identifier),
+        policySetName: defaultEmpty(policySet.name),
+        status: defaultEmpty(policySet.status),
+        identifier: defaultEmpty(policySet.identifier),
+        created: defaultEmpty(policySet.created),
+        accountId: defaultEmpty(policySet.account_id),
+        orgId: defaultEmpty(policySet.org_id),
+        projectId: defaultEmpty(policySet.project_id),
+        policyMetadata: map(policySet.details, policy => ({
+          id: id,
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts:39]: Not worth fixing, but...
```suggestion
          id,
```

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  iacmData: IacmFormattedData[]
+  iacmStatus: EvaluationStatus
+} => {
+  const iacmData: IacmFormattedData[] = []
+  let status: EvaluationStatus = EvaluationStatus.PASS
+  const statuses: EvaluationStatus[] = []
+  forEach(data, ({ id, details }) =>
+    forEach(details, policySet => {
+      statuses.push(defaultTo(policySet.status as EvaluationStatus, EvaluationStatus.PASS))
+      iacmData.push({
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts:29]: This just screams `reduce` to me, but it's fine as-is 👍🏻

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { defaultTo, forEach, includes, map } from 'lodash-es'
+import { GovernanceMetadata } from 'services/cd-ng'
+import { Evaluation } from 'services/pm'
+import { EvaluationStatus, IacmFormattedData } from './types'
+
+/* istanbul ignore next */
+const defaultEmpty = (value: string | number | undefined): string => defaultTo(value, '') as string
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/utils.ts:14]: You can probably simplify here with `unknown`
```suggestion
const defaultEmpty = (value: unknown): string => defaultTo(value, '') as string
```

Or if you're confident that `value` really will be either missing or a string...
```suggestion
const defaultEmpty = (value?: string): string => defaultTo(value, '')
```

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx [hunk 5]
```diff
@@ -11,6 +11,15 @@ import { TestWrapper } from '@common/utils/testUtils'
 import mockImport from 'framework/utils/mockImport'
 import ExecutionPolicyEvaluationsView from '../ExecutionPolicyEvaluationsView'
 
+jest.mock('services/pm', () => ({
+  useGetEvaluationList: jest.fn(() => ({
+    data: null,
+    loading: false,
+    refetch: jest.fn().mockReturnValue({}),
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/__tests__/ExecutionPolicyEvaluationsView.test.tsx:18]: Does this get called at all? If not, you can probably simplify
```suggestion
    refetch: jest.fn(),
```

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 6]
```diff
@@ -1,29 +1,61 @@
...
   const context = useExecutionContext()
   const governanceMetadata = context?.pipelineExecutionDetail?.pipelineExecutionSummary?.governanceMetadata
+  const queryParams = {
+    accountIdentifier: accountId,
+    orgIdentifier,
+    projectIdentifier,
+    includeChildScopes: true,
+    module: 'iacm',
+    entity: `iacm-${executionIdentifier}`,
+    lazy: !IACM_ENABLED
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:34]: If this is the `lazy` from Restful React, then it needs to be passed as a sibling to `queryParams`, not as a query param itself.
```typescript
const { data, loading, error } = useGetEvaluationList({
  queryParams,
  lazy: !IACM_ENABLED
})
```

You might find that this is being called regardless of the state of the flag

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 7]
```diff
@@ -1,29 +1,61 @@
+/* istanbul ignore file no unit tests as using MFE, tested with intregration tests */
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:1]: Naughty!

**Review by knagurski — DISMISSED** [review]
Some things to look at, but should be fine for a HF

**Comment by willd0yle** [comment]
> Some things to look at, but should be fine for a HF

I will create a pr with the improvements to merge into develop. Need to get this out ASAP

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx (hunk 6 above)
Comment (rameshx) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:34]: @willd0yle Is it okay if this API gets called when the flag isn't enabled? If not, can this be fixed in this PR?

**Review by rameshx — DISMISSED** [review]

**Pull request closed without merging**